File: go/isbn-verifier/isbn_verifier.go
package isbn

import (
	"strings"
	"unicode"
)

//IsValidISBN checks if the isbn is valid
func IsValidISBN(isbn string) bool {

	isbn = strings.ReplaceAll(isbn, "-", "")
	if len(isbn) != 10 {
		return false
	}

	sum := 0
	for i, s := range isbn {
		if s == 'X' && i == 9 {
			sum += 10
			continue
		}
		if !(unicode.IsDigit(s)) {
			return false
		}
		sum += int(s-'0') * (10 - i)
	}

	return sum%11 == 0
}

File: Exercism_scripts/python/isbn-verifier/isbn_verifier.py
def is_valid(isbn):
    valid = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "X"]
    isbn_repl = [x for x in isbn if x in valid]

    if len(isbn_repl) != 10 :
        return False

    index = 10
    sum = 0 
    for num in isbn_repl[:-1]:
        if num not in valid[:-1]:
            return False
        sum += int(num) * index
        index -= 1

    if isbn_repl[-1] in valid:
        if isbn_repl[-1] == "X":
            num = 10
            sum = sum + num * 1
        else:
            sum = sum + int(isbn_repl[-1]) * 1

    if int(sum) % 11 == 0:
        return True
    
    return False


